realm/realm-library/src/main/cpp/util.cpp

```diff
@@ -365,11 +365,29 @@ static string string_to_hex(const string& message, StringData& str, const char*
     return ret.str();
 }
 
+static string error_code_to_message(size_t error_code){
```

Comment:
Just a suggestion: the function name seems too generic/broad, given this util file is used all over the place. It could be renamed to something like `str_to_hex_error_code_to_message` ...